model/patch_lifecycle.go

```diff
@@ -491,6 +491,106 @@ func AbortPatchesWithGithubPatchData(createdBefore time.Time, closed bool, newPa
 }
 
+func MakeCommitQueueDescription(patches []patch.ModulePatch, project *Project) string {
+	commitFmtString := "'%s' into '%s/%s:%s'"
+	description := []string{}
+	for _, p := range patches {
+		owner := project.Owner
+		repo := project.Repo
+		branch := project.Branch
+		if p.ModuleName != "" {
+			module, err := project.GetModuleByName(p.ModuleName)
+			if err != nil {
+				continue
+			}
+			owner, repo = module.GetRepoOwnerAndName()
+			branch = module.Branch
+		}
+		commitMessages := make([]string, 0, len(p.PatchSet.Summary))
+		for _, summary := range p.PatchSet.Summary {
+			commitMessages = append(commitMessages, summary.Description)
+		}
+
+		description = append(description, fmt.Sprintf(commitFmtString, strings.Join(commitMessages, " <- "), owner, repo, branch))
+	}
+
+	if len(description) == 0 {
+		description = []string{"No Commits Added"}
+	}
+
+	return "Commit Queue Merge: " + strings.Join(description, " || ")
+}
+
+func MakeMergePatchFromExisting(existingPatchID string, u *user.DBUser) (*patch.Patch, error) {
+	existingPatch, err := patch.FindOne(patch.ById(patch.NewId(existingPatchID)))
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting existing patch '%s'", existingPatchID)
+	}
+	if existingPatch == nil {
+		return nil, errors.Errorf("patch '%s' doesn't exist", existingPatchID)
+	}
+
+	// verify the patch and its modules are in mbox format
+	if !existingPatch.CanEnqueue() {
+		return nil, errors.Errorf("can't enqueue non-mbox patch '%s'", existingPatchID)
+	}
+
+	// verify the commit queue is on
+	projectRef, err := FindOneProjectRef(existingPatch.Project)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't get project ref '%s'", existingPatch.Project)
+	}
+	if projectRef == nil {
+		return nil, errors.Errorf("no project '%s' exists", existingPatch.Project)
+	}
+	if !projectRef.Enabled {
+		return nil, errors.Errorf("project '%s' is disabled", existingPatch.Project)
+	}
+	if projectRef.PatchingDisabled {
+		return nil, errors.Errorf("patching is disabled for project '%s'", existingPatch.Project)
+	}
+	if !projectRef.CommitQueue.Enabled {
+		return nil, errors.Errorf("commit queue is disabled for project '%s'", existingPatch.Project)
+	}
+
+	project := &Project{}
+	if _, err = LoadProjectInto([]byte(existingPatch.PatchedConfig), existingPatch.Project, project); err != nil {
+		return nil, errors.Wrap(err, "problem loading project")
+	}
+
+	patchDoc := &patch.Patch{
+		Id:            mgobson.NewObjectId(),
+		Description:   MakeCommitQueueDescription(existingPatch.Patches, project),
+		Author:        u.Username(),
+		Project:       existingPatch.Project,
+		Githash:       existingPatch.Githash,
+		Status:        evergreen.PatchCreated,
+		Alias:         evergreen.CommitQueueAlias,
+		Patches:       existingPatch.Patches,
+		PatchedConfig: existingPatch.PatchedConfig,
+		CreateTime:    time.Now(),
+		DisplayNewUI:  existingPatch.DisplayNewUI,
```

Comment:
This field won't do anything - it's to tell the CLI to append a spruce redirect to the URL

Reply:
removed